### PR TITLE
New package: ibus-chewing

### PR DESCRIPTION
**Pull request by funk443**
#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

#### Local build testing
- I built this PR locally for my native architecture, x86_64-glibc

### CODE REVIEW

File: srcpkgs/ibus-array/template [hunk 1]
```diff
@@ -1,10 +1,11 @@
 # Template file for 'ibus-array'
 pkgname=ibus-array
 version=0.2.2
-revision=1
+revision=2
 build_style=gnu-configure
 hostmakedepends="automake pkg-config python3 gettext-devel libtool"
 makedepends="ibus-devel gettext-devel sqlite-devel"
+depends="ibus"
```
Comment (oreo639) [inline srcpkgs/ibus-array/template:8]: ibus-array already depends on ibus, you can confirm using `xbps-query -Rx ibus-array`.
(it should be added in the "collect rundeps" step)

File: srcpkgs/ibus-chewing/template [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+hostmakedepends="git cmake pkg-config"
+makedepends="gettext-devel libchewing-devel ibus-devel gob2 libX11-devel gtk+3-devel glib-devel findutils"
+depends="ibus libchewing"
+short_desc="Chewing input method for ibus"
+maintainer="ctoid <funk443@yandex.com>"
+license="GPL-2.0-or-later"
+homepage="https://github.com/definite/ibus-chewing"
+
+do_fetch() {
+	git clone https://github.com/definite/ibus-chewing.git $wrksrc
```
Comment (oreo639) [inline srcpkgs/ibus-chewing/template:16]: We don't generally accept packages using git clone. We definitely will not accept a package using git and cloning straight to the master branch.

Please use tagged releases whenever possible, and use proper distfiles whenever possible.

File: srcpkgs/ibus-chewing/template [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+pre_configure() {
+	rm -rf cmake-fedora
+	git clone https://pagure.io/cmake-fedora.git
+}
+
+do_configure() {
+	cmake . -DCMAKE_INSTALL_PREFIX='/usr' -DLIBEXEC_DIR='/usr/libexec'
```
Comment (oreo639) [inline srcpkgs/ibus-chewing/template:25]: For this, you shouldn't need to add a custom do_configure, you should be able to just do `configure_args`.

File: srcpkgs/ibus-chewing/template [hunk 4]
```diff
@@ -0,0 +1,30 @@
+# Template file for 'ibus-chewing'
+pkgname=ibus-chewing
+version=1.5.1
+revision=1
+archs="x86_64"
```
Comment (oreo639) [inline srcpkgs/ibus-chewing/template:5]: Why is this restricted to x86_64? (if, for example, it doesn't cross compile, it should instead be marked as nocross)

File: srcpkgs/ibus-chewing/template [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+build_style=gnu-makefile
+hostmakedepends="git cmake pkg-config"
+makedepends="gettext-devel libchewing-devel ibus-devel gob2 libX11-devel gtk+3-devel glib-devel findutils"
```
Comment (oreo639) [inline srcpkgs/ibus-chewing/template:8]: `makedepends` is for library dependencies, `hostmakedepends` is for application dependencies, `gettext-devel` should be in `hostmakedepends`, I imagine `findutils` should be as well.

**Review by oreo639 — COMMENTED** [review]
One commit per package, and no merge commits.
https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#creating-updating-and-modifying-packages-in-void-by-yourself

You can get rid of an existing commit using `git reset HEAD~` or using `git rebase -i`.

File: srcpkgs/ibus-chewing/template (hunk 3 above)
Comment (classabbyamp) [reply srcpkgs/ibus-chewing/template:25]: and `-DCMAKE_INSTALL_PREFIX='/usr'` is already defined in the build-style so it doesn't need to specified.

File: srcpkgs/ibus-chewing/template [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+	git clone https://pagure.io/cmake-fedora.git
+}
+
+do_configure() {
+	cmake . -DCMAKE_INSTALL_PREFIX='/usr' -DLIBEXEC_DIR='/usr/libexec'
+}
+
+post_install() {
+	glib-compile-schemas /usr/share/glib-2.0/schemas
+}
```
Comment (sgn) [inline srcpkgs/ibus-chewing/template:30]: Please don't do this.

File: srcpkgs/ibus-chewing/template [hunk 7]
```diff
@@ -3,28 +3,22 @@ pkgname=ibus-chewing
...
-do_fetch() {
-	git clone https://github.com/definite/ibus-chewing.git $wrksrc
-}
+distfiles="https://github.com/definite/ibus-chewing/archive/refs/tags/${version}.tar.gz"
+checksum="4c0b58e9b13b826969d8df6276275bbd893a4a6d665cc218a65aa3eecde00d53"
+nocross=yes
 
 pre_configure() {
 	rm -rf cmake-fedora
 	git clone https://pagure.io/cmake-fedora.git
```
Comment (classabbyamp) [inline srcpkgs/ibus-chewing/template:22]: why is this needed?

Comment (funk443) [reply srcpkgs/ibus-chewing/template:22]: You mean the cmake-fedora part?

Comment (Duncaen) [reply srcpkgs/ibus-chewing/template:23]: Yes, if the project actually depends on a specific git submodule, we generally add new distfile (tar.gz for a specific commit) to it and move it into the right place.

Comment (funk443) [reply srcpkgs/ibus-chewing/template:23]: I added this part because the original cmake-fedora host is no longer available.

File: srcpkgs/ibus-chewing/template [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+pkgname=ibus-chewing
+version=1.5.1
+revision=1
+archs="x86_64"
+build_wrksrc=ibus-chewing-1.5.1
+build_style=cmake
+configure_args="-DLIBEXEC_DIR=/usr/libexec"
+make_cmd=make
+hostmakedepends="git cmake pkg-config findutils gettext-devel"
+makedepends="libchewing-devel ibus-devel gob2 libX11-devel gtk+-devel glib-devel"
```
Comment (oreo639) [inline srcpkgs/ibus-chewing/template:11]: I didn't notice this earlier, sorry, but I don't believe new gtk2 packages will be accepted.

Comment (funk443) [reply srcpkgs/ibus-chewing/template:11]: Ya, that's why I directly cloned the repo before, instead of downloading the realeases.

Comment (paper42) [reply srcpkgs/ibus-chewing/template]: You can get a tarball at a specific commit from GitHub. I don't think this meets quality requirements, it either uses GTK2 or it doesn't have a release.

File: srcpkgs/ibus-chewing/template (hunk 4 above)
Comment (paper42) [reply srcpkgs/ibus-chewing/template:5]: This was not resolved, please don't mark it resolved

File: srcpkgs/ibus-chewing/template (hunk 8 above)
Comment (funk443) [reply srcpkgs/ibus-chewing/template]: I just simply fork it, and release the latest commit by myself.

Comment (paper42) [reply srcpkgs/ibus-chewing/template]: No, that's equivalent to just not using a release. A release is required because the developers know the best when the code is stable and usable. If we packaged the latest commit, we would have to figure out issues on our own and the maintenance burden would be much higher.

Comment (funk443) [reply srcpkgs/ibus-chewing/template]: Oh, ok then...  What should I do?

Comment (paper42) [reply srcpkgs/ibus-chewing/template]: It's not acceptable right now.

Comment (classabbyamp) [reply srcpkgs/ibus-chewing/template]: you could ask upstream to make a release

Comment (funk443) [reply srcpkgs/ibus-chewing/template]: All right then, sorry for wasting you guys so much time.
